packages/require/require.js

```diff
@@ -131,4 +132,29 @@ const Require = {
   }
 };
 
+function _readSource(file) {
+  const source = fs.readFileSync(file, { encoding: "utf8" });
+  const fileExtension = path.extname(file).toLocaleLowerCase();
+  const isTs = [".ts", ".tsx"].includes(fileExtension.toLocaleLowerCase());
+  if (isTs) {
+    try {
+      require("ts-node");
+    } catch {
+      throw new TruffleError(`Attempted to execute script with extension
+        ${fileExtension}, but the 'ts-node' module, or one of its required peers
+        has not been installed.`);
+    }
```

Comment:
This is necessary because `ts-node` is an optional peer dependency, and people who aren't using TS shouldn't need to install it.